cfpb/consumerfinance.gov
# Schema module updates

As part of sunsetting the auto loan tool, we need to add enhancements to a few Ask modules to enable migration of some  tool content to Ask questions.
- __Jump links__: The design for the enhanced Ask questions includes jump links to sections of the page. To enable that, this PR adds anchor tag fields to the `Text`, `How To`, and `FAQ` modules on answer pages
- __How to schema updates__: The design for the enhanced Ask questions includes the application of How To schema markup to top-level sections of the answer page with H2 headers. To enable that, this PR adds a few fields to the `How To` module:
    - a tag option for the How To title (h2, h3, h4)
    - a toggle for the visual display of the How To title in the page
    - a tag option for the step titles (h2, h3, h4, b, and p)
    - a toggle for display of numbers before step titles

## Additions

- __How To schema options__, including title and step title tag dropdowns & number and title display toggles
- __Anchor tag fields__ on `FAQ`, `How To`, and `Text` modules


## Changes
- updates to answer and schema templates to accommodate new options
- updates to schema less to handle How To spacing on blog pages
- updates to how to schema test


## How to test this PR

1. Check out branch and run `yarn styles` and migrations
2. Verify that the two existing pages with HowTo schema ([page 1](http://localhost:8000/about-us/blog/credit-card-debt-during-coronavirus-relief-options-tips/) and [page 2](http://localhost:8000/about-us/blog/tarjeta-de-credito-y-coronavirus-opciones-de-alivio/)) still display acceptably
3. Verify that the anchor tag fields work on the How To, FAQ, and Text modules
4. Verify that the new fields on the How To module (title and step title tag options, title and number display toggles) work as expected
5. Validate the HowTo schema output by testing the source code with the [schema.org validator](https://validator.schema.org/)

## Screenshots

### Updated How To schema admin
| ![how-to-admin](https://user-images.githubusercontent.com/778171/198356913-6a06ac93-0695-433b-89c4-5e946a4836ef.png) |
| -- |



## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
